src/main/java/com/amazon/opendistroforelasticsearch/security/resolver/IndexResolverReplacer.java

```diff
@@ -351,53 +349,61 @@ public Resolved resolveRequest(final Object request) {
 
     public final static class Resolved implements Serializable, Writeable {
 
-        /**
-         *
-         */
-        private static final Set<String> All_SET = Collections.singleton("*");
+        private static final Set<String> All_SET = ImmutableSet.of("*");
```

Comment:
@sujithvm I understand this PR has been merged. But out of curiosity, why replaced `Collections.singleton()` with `ImmutableSet.of()`? Does it improve performance?